src/tests/java/picard/analysis/CollectGcBiasMetricsTest.java

```diff
@@ -176,36 +124,193 @@ public void runTest(final File input) throws IOException {
+        final File sortedSam = File.createTempFile("CollectGcBias", ".bam", TEST_DIR);
+
+        final SAMFileWriter writer = new SAMFileWriterFactory()
+                .setCreateIndex(true).makeBAMWriter(header, false, unsortedSam);
+
+        for( final SAMRecordSetBuilder subSetBuilder : setBuilder){
+            for (final SAMRecord record : subSetBuilder) {
+                writer.addAlignment(record);
+            }
+        }
+        writer.close();
+
+        final SortSam sorter = new SortSam();
+        final String[] args = new String[] {
+                "INPUT=" + unsortedSam.getAbsolutePath(),
+                "OUTPUT=" + sortedSam.getAbsolutePath(),
+                "SORT_ORDER=coordinate"
+        };
+
+        sorter.instanceMain(args);
+
+        return sortedSam;
+    }
+
+    /////////////////////////////////////////////////////////////////////////////
+    // Runs CollectGcBias with input Sam file and outputs details and summary files for truth assertion.
+    /////////////////////////////////////////////////////////////////////////////
+    public void runGcBias (final File input, final File outfile, final File detailsOutfile) throws IOException {
+        final String referenceFile = "testdata/picard/metrics/chrMNO.reference.fasta";
+        final File pdf = File.createTempFile("test", ".pdf");
+        pdf.deleteOnExit();
+
+        final int windowSize = 100;
+        final double minGenFraction = 1.0E-5;
+        final boolean biSulfiteSeq = false;
+        final boolean assumeSorted = false;
+
+        final String[] args = new String[]{
+                "INPUT=" + input.getAbsolutePath(),
+                "OUTPUT=" + detailsOutfile.getAbsolutePath(),
+                "REFERENCE_SEQUENCE=" + referenceFile,
+                "SUMMARY_OUTPUT=" + outfile.getAbsolutePath(),
+                "CHART_OUTPUT=" + pdf.getAbsolutePath(),
+                "SCAN_WINDOW_SIZE=" + windowSize,
+                "MINIMUM_GENOME_FRACTION=" + minGenFraction,
+                "IS_BISULFITE_SEQUENCED=" + biSulfiteSeq,
+                "LEVEL=ALL_READS",
+                "LEVEL=SAMPLE",
+                "LEVEL=READ_GROUP",
+                "ASSUME_SORTED=" + assumeSorted
+        };
+        Assert.assertEquals(runPicardCommandLine(args), 0);
+    }
+
+    /////////////////////////////////////////////////////////////////////////////
+    //Used to generate the Sam Record Sets with SamRecordSetBuilder.addPair().
+    //testNumber 1: runGcBiasMultiLevelTest, generates records aligning to chrM and chrO
+    //testNumber 2: runWindowsComparisonTest, generates records aligning to chrM,N,O.
+    /////////////////////////////////////////////////////////////////////////////
+    public void setup(final int ID, final String readGroupId, final SAMReadGroupRecord readGroupRecord, final String sample,
+                      final String library, final SAMFileHeader header, final SAMRecordSetBuilder setBuilder, final int testNumber)
```

Comment:
Instead of passing in ```testNumber```, I would just make a separate setup function for each test.